tests/test_converter.py

```diff
@@ -1,12 +1,35 @@
 import unittest
+from tests.java.java_server import JavaServer
+
+from py4j.java_gateway import JavaGateway
 
 from patternconverter import PatternConverter
 
 
 class ConverterTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.java_gateway_server = JavaServer()
+        cls.java_gateway_server.run_server()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.java_gateway_server.shutdown_server()
 
     def setUp(self):
-        pass
+        self.gateway = JavaGateway()
+
+    def test_year_month_day_format_conversion_to_java(self):
+        date = "2018-11-07"
+
+        python_format = "%Y-%M-%d"
+        c = PatternConverter()
+        converted_format = c.convert_to_iso_8601(pattern=python_format)
+
+        dt_formatter = self.gateway.jvm.java.time.format.DateTimeFormatter
```

Comment:
:1st_place_medal:

Reply:
This is actually always gonna be the java code that we are running, so can be nice if we live the java code to the java process (besides calling a function with this code)

Reply:
Hmmmm interesting idea. We can definitely do that. I'll push it together with the fixes I have for the Java server code monitoring. 
On the meanwhile you can just abstact that by extracting a function that, let's say, gets a format and a date str and returns the formatted date str (?)...

Reply:
@Iddoyadlin we now have an entry point to the java server which means we can call the methods in that class with the py4j :)
The java class still might need a little work but the example test now works with it.